Sources/Sentry/SentrySessionReplaySyncC.c

```diff
@@ -0,0 +1,93 @@
+#include "SentrySessionReplaySyncC.h"
+#include "SentryAsyncSafeLog.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static SentryCrashReplay crashReplay = { 0 };
```

Comment:
The important code that uses this during the crash cannot be synchronized.
This won't be a problem.

Reply:
I was looking at SentrySessionReplayIntegration's methods, not the crash reporter:
- sentrySessionReplaySync_start called by saveCurrentSessionInfo which originates from:
    - -[SentrySessionReplayIntegration newSceneActivate] from UISceneDidActivateNotification
    - -[SentrySessionReplayIntegration sentrySessionStarted:] from SentrySDK.startSession
- sentrySessionReplaySync_updateInfo called by sessionReplayNewSegmentWithReplayEvent, which originates from:
    - +[PrivateSentrySDKOnly captureReplay]
    - -[SentrySessionReplayIntegration installWithOptions:]
    - SentrySessionReplay.newFrame(_:) from CADisplayLink
    - -[SentrySessionReplayIntegration sentrySessionEnded:] from SentrySDK.endSession
    - -[SentrySessionReplayIntegration uninstall]
    - SentrySessionReplay.newFrame(_:)
- sentrySessionReplaySync_readInfo called by resumePreviousSessionReplay, which originates from:
    - -[SentrySessionReplayIntegration installWithOptions:]

What happens if SentrySDK.startSession is called while a replay has already fired a couple calls to newFrame?

Reply:
Also, I'm not super well-versed in the UIScene API; since we observe `UISceneDidActivateNotification`, what happens if an app has two different scenes visible while split-screen multitasking on iPadOS and both are recording replays?

Reply:
> What happens if SentrySDK.startSession is called while a replay has already fired a couple calls to newFrame?

We should send the old replay and start a new one. But Im noticed Im just invalidating the old reply and not sending it.

> what happens if an app has two different scenes visible while split-screen multitasking on iPadOS and both are recording replays?

We listen to the first activatedScene and ignore the others.